app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,45 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      providers = {
+        :gce => [ManageIQ::Providers::Google::CloudManager, :name],
+        :aws => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      instance_uri = URI(provider_id)
+      prov, name_field = providers[instance_uri.scheme.to_sym]
```

Comment:
@enoodle to be safer I wouldn't try to convert this `to_sym`, in fact a `nil` scheme would crash here. Keep it as string, performance is the same. (You obviously have to change the `providers` hash keys as well).